src/main/scala/calc/Compiler.scala

```diff
@@ -61,21 +62,113 @@ object Compiler {
+      "pow" -> 2,
+      "abs" -> 1,
+      "max" -> 2,
+      "min" -> 2)
 
     tree match {
       case Literal(value) =>
         irt.DoubleLiteral(value)
 
-      case _ =>
-        throw new Exception(
+      case BinaryOp(op, lhs, rhs) =>
+        val jsop = op match {
+          case "+" => irt.BinaryOp.Double_+
+          case "-" => irt.BinaryOp.Double_-
+          case "*" => irt.BinaryOp.Double_*
+          case "/" => irt.BinaryOp.Double_/
+        }
+        irt.BinaryOp(jsop, compileExpr(lhs, typeEnv), compileExpr(rhs, typeEnv))
+
+      case Let(ident, value, body) =>
+        val jsV = compileExpr(value, typeEnv)
+        val jsIdent = irt.Ident(ident.name)
+        val jsId = irt.VarDef(jsIdent, jsV.tpe, mutable = false, jsV)
+        irt.Block(List(jsId, compileExpr(body, typeEnv + (ident.name -> jsV.tpe))))
+
+      case Ident(n) =>
+        try {
+          irt.VarRef(irt.Ident(n))(typeEnv(n))
+        }catch{
+          case e:Exception => throw new Exception("No such identitier!")
+        }
+
+      case If(cond, thenp, elsep) =>
+        val jsThenp = compileExpr(thenp, typeEnv)
+        val jsElsep = compileExpr(elsep, typeEnv)
+        val jsCond = compileExpr(cond, typeEnv)
+        val finalCond = irt.BinaryOp(irt.BinaryOp.!==, jsCond, irt.DoubleLiteral(0.0))
+        if (jsElsep.tpe != jsElsep.tpe){
+          throw new Exception(s"The then and also phase in the conditional statement must bse same type")
+        }
+        irt.If(finalCond, jsThenp, jsElsep)(jsThenp.tpe)
+
+      case Closure(idList, body) =>
+        val captureList = findCapture(tree).toList
+        val captureParam = captureList.map(s =>
+          irt.ParamDef(irt.Ident(s), typeEnv(s), mutable = false, rest = false))
+        val params = idList.map(s =>
+          irt.ParamDef(irt.Ident("$$"+s.name), irtpe.AnyType, mutable = false, rest = false))
+        val newTypeEnv = idList.foldLeft(typeEnv)((env: Map[String, irtpe.Type], s:Ident)
+                                                    => env+(s.name -> irtpe.DoubleType))
+        val unboxParams = idList.map(s =>
+          irt.VarDef(irt.Ident(s.name), irtpe.DoubleType, mutable = false,
+            irt.Unbox(irt.VarRef(irt.Ident("$$"+s.name))(irtpe.AnyType), 'D')))
+        val bl = irt.Block(unboxParams ++ List(compileExpr(body, newTypeEnv)))
+        irt.Closure(captureParam, params, bl, captureList.map(s => irt.VarRef(irt.Ident(s))(typeEnv(s))))
+
+      case Call(f, args) =>
+        try {
+          val jsFunction = compileExpr(f, typeEnv)
+          irt.Unbox(irt.JSFunctionApply(jsFunction,args.map(s => compileExpr(s, typeEnv))), 'D')
+        }catch{
+          case e:Exception=> {
+            try {
+              val funcName = f.asInstanceOf[Ident].name
+              val identName = funcName + "__D" * functionTypes(funcName) + "__D"
+              irt.Apply(mathModule, irt.Ident(identName),
+                args.map(s => compileExpr(s, typeEnv)))(irtpe.DoubleType)
```

Comment:
Basically, here you could just write `irt.LoadModule(irtpe.ClassType("jl_Math$"))` insted of `mathModule`, and be done with it.